### PR TITLE
feat(core): Add lua modules

### PR DESCRIPTION
**Pull request by dranikpg**
Add the following modules to the lua interpreter:
* cjson
* cmsgpack
* struct
* bit

and registers them in the globals table.

Helps with #184

### CODE REVIEW

**Comment by dranikpg** [comment]
I've chosen to embed the module sources at `src/redis/lua` because:
- the build process of cjson/cmsgpack would have to be patched significantly
- bit/struct are standalone and would require a build solution on their own

I guess, a single target with a single build file at `src/redis/lua` would be enough? The other option would be fully subdividing everything.

**Comment by romange** [comment]
Looks good. 
1. A single target is fine.
2. We need tests inside interpreter_test that cover the functionality of all modules
3. If you had to change the original files - please add a readme explaining in a few lines what had to be changed or provide a patch snippets (whatever is simpler).

**Comment by dranikpg** [comment]
I've patched the registration process, because lua 5.2 changed the way modules are registered. (`lual_register` became deprecated https://www.lua.org/manual/5.2/manual.html, the new `luaL_newlib` doesn't make it global upon registration to be used with `require` instead)

The modules show quite a lot of warnings (about 20) during compile (mainly sign-compare, implicit-fallthrough, misleading-identation). Should I fix them or just suppress them in the sub-build?

**Comment by romange** [comment]
just opress them - see target_compile_options in redis CMakeLists

**Comment by romange** [comment]
do you mind squashing the commits into a single one?

**Review by romange — APPROVED** [review]

File: src/redis/lua/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+add_definitions(-DENABLE_CJSON_GLOBAL)
```
[the remark this answers is not in the document]
Comment (dranikpg) [reply src/redis/lua/CMakeLists.txt]: I just removed `add_definitions` together with the ifdef in the header

**Pull request merged**